### PR TITLE
Scalar export

### PR DESCRIPTION
**Pull request by c0cky**
This export function allows MT to connect with scalar and exporting the annotations.

The user will be logged into their class and the annotations from their class will be available in rdf-json format for scalar at the url of http://mediathreadinstall.com/asset/scalar

The user may then use this information and plug it into scalar.
mediathread-to-scalar-viz-2

### CODE REVIEW

File: mediathread/assetmgr/views.py [hunk 1]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+            ar = AssetResource(include_annotations=True)
+            ar.Meta.excludes = ['added', 'modified', 'course', 'active']
+            notes = SherdNote.objects.get_related_notes(
+                                                        assets,
+                                                        course_mem['id'],
+                                                        [course_mem['id']],
+                                                        True
+                                                        )
+
+            api_response += (ar.render_list(request, [course_mem['id']],
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: The code here renders the items & notes to json. Then, at Ln 687, the json is rendered into the scalar export format. The scalar export could probably work directly against the item/note objects proper. What's the reasoning behind the interim json step?

File: mediathread/assetmgr/views.py [hunk 2]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        for n in range(0, data.get('annotation_count')):
+            self.parse_annotations(data, n)
+
+    def get(self, request):
+        self.root = 'http://' + request.get_host()
+        assets = Asset.objects.filter(course=request.course)
+        course = Course.objects.get(id=request.course.id)
+        api_response = []
+        n = 0
+        for course_mem in course.user_set.values():
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: Is it necessary to iterate by course member? Would be much simpler to just iterate the assets and spit out the scalar equivalent.

File: mediathread/assetmgr/views.py [hunk 3]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+             data.get('annotations')[n]['url'].rstrip('/'),
+             "type": "uri"}]
+        time = ''
+        time += self.root
+        time += data.get('local_url').rstrip('/')
+        time += '#t=npt:'
+        time += str(data.get('annotations')[n]['annotation']['start'])
+        time += ','
+        time += str(data.get('annotations')[n]['annotation']['end'])
+        # try:
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: commented code should be removed.

File: mediathread/assetmgr/views.py [hunk 4]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        a_annotation_node[
+            'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+        ] = [
+            {"value": "http://www.openannotation.org/ns/Annotation",
+             "type": "uri"}]
+        anno_urn = 'urn:mediathread:anno' + str(self.anno_num + 1)
+        self.export[anno_urn.rstrip('/')] = a_annotation_node
+        self.anno_num += 1
+
+        tag = []
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: redundant declaration

File: mediathread/assetmgr/views.py [hunk 5]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        tag = (data.get('annotations')[n]['metadata']['tags'])
+        for k in range(0, len(tag)):
+            self.parse_tag(k, tag, n)
+
+        vocab = []
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: redundant declaration

File: mediathread/assetmgr/views.py (hunk 1 above)
Comment (c0cky) [inline mediathread/assetmgr/views.py]: I had become familiar with the json format, which exports everything neatly, while trying this previously with xml.

File: mediathread/assetmgr/views.py (hunk 2 above)
Comment (c0cky) [inline mediathread/assetmgr/views.py]: Does the assets have all of the information for every users annotations in the class?

Comment (sdreher) [inline mediathread/assetmgr/views.py]: Yes. You can get all course annotations like so: SherdNote.objects.filter(asset__course=my_course)

File: mediathread/assetmgr/views.py [hunk 6]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        self.parse_vocab(vocab, n, data)
+
+        try:
+            self.export[self.root + data.get('annotations')[n]
+                        ['url'].rstrip('/')] = annotation_node
+        except Exception:
+            pass
+
+    def parse_vocab(self, vocab, n, data):
+        print vocab
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: print statements should be removed

File: mediathread/assetmgr/views.py [hunk 7]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        for j in range(0, len(vocab)):
+                num = 0
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: I think the indentation is doubled here - 8 rather than 4

Comment (c0cky) [inline mediathread/assetmgr/views.py]: That is one line that is indented to conform to pep8. I can create a variable to shorten the line.

File: mediathread/assetmgr/views.py [hunk 8]
```diff
@@ -448,6 +450,245 @@ def final_cut_pro_xml(request, asset_id):
...
+        a_annotation_node = {}
+        a_annotation_node['http://www.openannotation.org/ns/hasBody'] = [
+            {"value": self.root +
+             data.get('annotations')[n]['url'].rstrip('/'),
+             "type": "uri"}]
+        time = ''
+        time += self.root
+        time += data.get('local_url').rstrip('/')
+        time += '#t=npt:'
+        time += str(data.get('annotations')[n]['annotation']['start'])
```
Comment (sdreher) [inline mediathread/assetmgr/views.py]: annotations may or may not have start/end properties

**Comment by sdreher** [comment]
Merged into https://github.com/ccnmtl/mediathread/tree/dartmouth-scalar-export

**Pull request closed without merging**